File: ParcialWeb/ClientApp/src/app/vivirmejor/models/cliente.ts
export class Cliente {

    identificacion: string;
    nombre: string;
    proyecto : string;
    valor : number;
    cuotaInicial : number;
    abono : number;

    calcularValor(){
        switch (this.proyecto){
            case 'BM':
                this.valor = 40000.00;
                this.cuotaInicial = this.valor * 0.2;
                this.abono = this.cuotaInicial;
            break;   
            case 'BC':
                this.valor = 50000.00;
                this.cuotaInicial = this.valor * 0.2
                this.abono = this.cuotaInicial;

            break; 
            case 'JR':
                this.valor = 70000.00;
                this.cuotaInicial = this.valor * 0.2
                this.abono = this.cuotaInicial;

            break; 
        }
    }

}

File: Entity/Cliente.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Entity
{
    public class Cliente
    {
        [Key]
        public string Identificacion { get; set; }
        public string Nombre { get; set; }

        [Required]
        public int CodProyecto { get; set; }
        public ProyectoVivienda Proyecto { get; set; }

        public ICollection<Abono> Abonos { get; set; }
    }
}
